File: PreUtil/src/output/c/easy/RansomNote.c
/**https://leetcode.com/problems/ransom-note/ */
//Given two strings ransomNote and magazine, return true if ransomNote can be constructed by using the letters from magazine and false otherwise.
//Each letter in magazine can only be used once in ransomNote.


bool canConstruct(char * ransomNote, char * magazine){
    int len1 = strlen(ransomNote);
    int len2 = strlen(magazine);
    if(len1 > len2){
        return false;
    }
    int arr[26] = {0};
    for(int i = 0; i < len2; i++){
        arr[magazine[i] - 'a']++;
    }
    for(int i = 0; i < len1; i++){
        arr[ransomNote[i] - 'a']--;
        if(arr[ransomNote[i] - 'a'] < 0){
            return false;
        }
    }
    return true;

}

    

File: Tool/PreUtil/src/output/java/easy/RansomNote.java
/**https://leetcode.com/problems/ransom-note/ */
//Given two strings ransomNote and magazine, return true if ransomNote can be constructed by using the letters from magazine and false otherwise.
//Each letter in magazine can only be used once in ransomNote.


class RansomNote {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] count = new int[26];
        for(int i = 0; i < magazine.length(); i++){
            count[magazine.charAt(i) - 'a']++;
        }
        for(int i = 0; i < ransomNote.length(); i++){
            count[ransomNote.charAt(i) - 'a']--;
            if(count[ransomNote.charAt(i) - 'a'] < 0) return false;
        }
        return true;
        
    }
}

    